File: app/src/main/java/com/example/group_project/controller/ReviewAdapter.java
package com.example.group_project.controller;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.group_project.R;
import com.example.group_project.model.Review;
import com.example.group_project.util.SpotUiFormatter;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class ReviewAdapter extends RecyclerView.Adapter<ReviewAdapter.ReviewViewHolder> {

    private final List<Review> reviewList;

    public ReviewAdapter(List<Review> reviewList) {
        this.reviewList = reviewList == null ? new ArrayList<>() : reviewList;
    }

    @NonNull
    @Override
    public ReviewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_review, parent, false);
        return new ReviewViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ReviewViewHolder holder, int position) {
        Review currentReview = reviewList.get(position);
        holder.spotNameText.setText(currentReview.getSpotName());
        holder.ratingText.setText(String.format(
                Locale.US,
                "Rating: %.1f/5",
                currentReview.getStarRating()
        ));
        holder.descriptionText.setText(currentReview.getDescription());
        holder.timestampText.setText(SpotUiFormatter.formatRelativeTime(currentReview.getTimestamp()));
    }

    @Override
    public int getItemCount() {
        return reviewList.size();
    }

    public void setReviews(List<Review> reviews) {
        reviewList.clear();
        if (reviews != null) {
            reviewList.addAll(reviews);
        }
        notifyDataSetChanged();
    }

    public static class ReviewViewHolder extends RecyclerView.ViewHolder {

        TextView spotNameText;
        TextView ratingText;
        TextView timestampText;
        TextView descriptionText;

        public ReviewViewHolder(@NonNull View itemView) {
            super(itemView);
            spotNameText = itemView.findViewById(R.id.textSpotName);
            ratingText = itemView.findViewById(R.id.textRatingSummary);
            timestampText = itemView.findViewById(R.id.textTimestamp);
            descriptionText = itemView.findViewById(R.id.textReviewBody);
        }
    }
}

File: app/src/main/java/com/example/group_project/controller/ReviewAdapter.kt
package com.example.group_project.controller

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.group_project.R
import com.example.group_project.model.Review
import com.example.group_project.util.SpotUiFormatter
import java.util.Locale

class ReviewAdapter(reviewList: List<Review>?) : RecyclerView.Adapter<ReviewAdapter.ReviewViewHolder>() {

    private val reviewList: MutableList<Review> = reviewList?.toMutableList() ?: ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReviewViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_review, parent, false)
        return ReviewViewHolder(view)
    }

    override fun onBindViewHolder(holder: ReviewViewHolder, position: Int) {
        val currentReview = reviewList[position]
        holder.spotNameText.text = currentReview.spotName
        // Keep rating display consistent across all review lists
        holder.ratingText.text = String.format(
            Locale.US,
            "Rating: %.1f/5",
            currentReview.starRating
        )
        holder.descriptionText.text = currentReview.description
        holder.timestampText.text = SpotUiFormatter.formatRelativeTime(currentReview.timestamp)
    }

    override fun getItemCount(): Int = reviewList.size

    fun setReviews(reviews: List<Review>?) {
        // Simple full refresh is fine for these small class-project lists
        reviewList.clear()
        if (reviews != null) {
            reviewList.addAll(reviews)
        }
        notifyDataSetChanged()
    }

    class ReviewViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val spotNameText: TextView = itemView.findViewById(R.id.textSpotName)
        val ratingText: TextView = itemView.findViewById(R.id.textRatingSummary)
        val timestampText: TextView = itemView.findViewById(R.id.textTimestamp)
        val descriptionText: TextView = itemView.findViewById(R.id.textReviewBody)
    }
}
